repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/herp/HerpUserAccessEventFilterTest.groovy

```diff
@@ -446,53 +443,109 @@ class HerpUserAccessEventFilterTest extends ReposeValveTest {
         "200"        | "test12" | "tenantId=12345" | "PUT"  | "OK"
     }
 
+    @Unroll
+    def "will populate and log the username '#expectedUsername' when the request header username is '#requestUsername' and the response header is '#responseUsername'"() {
+        given:
+        def requestHeaders = requestUsername ? [(OpenStackServiceHeader.USER_NAME.toString()): requestUsername] : [:]
+        def responseHeaders = responseUsername ? [(OpenStackServiceHeader.USER_NAME.toString()): responseUsername] : [:]
```

Comment:
This needs to be an explicit null check since I test empty strings which are "falsy" in Groovy.